### PR TITLE
ExtractURL

### PR DESCRIPTION
**Pull request by dem-evgeny**
Added the support of regex.
Added to the script:
1. Added function: checkRegex
2. Changed the all the "while (found = reg.exec(text))"

### CODE REVIEW

File: Scripts/script-ExtractURL.yml [hunk 1]
```diff
@@ -12,8 +17,17 @@ script: |-
...
   while (found = reg.exec(text)) {
-      if (filtered.indexOf(found[0]) < 0 && whitelist.indexOf(found[0]) < 0) {
-          matches[found[0]] = true;
+      matches[found[0]] = true;
+      for (var i = 0 ; i < whitelist.length; i++) {
+          if (whitelist[i] === found[0]) {
+              delete matches[found[0]];
+              break;
+          }else{
+              if(whitelist[i].match("^~") && checkRegex(whitelist[i],found[0])){
```
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: Move this condition to the checkRegex function.
and check for starts with instead

File: Scripts/script-ExtractURL.yml [hunk 2]
```diff
@@ -3,6 +3,11 @@ commonfields:
   version: -1
 name: ExtractURL
 script: |-
+  function checkRegex(reg,str){
```
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: In most cases we define the function as a variable:
```
var checkRegex = function() {
};
```

But not critical

File: Scripts/script-ExtractURL.yml (hunk 1 above)
Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: Changed except "startsWith()". Demisto doesn't recognise this command. I am getting this error: script failed to run: TypeError: 'startsWith' is not a function. So I left with the .match("^~") command.

File: Scripts/script-ExtractURL.yml (hunk 2 above)
Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: Changed

File: Scripts/script-ExtractURL.yml (hunk 1 above)
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: IndexOf('~') === 0

Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: changed

Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: Cant see your change ...

Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: Now you supposed to see it...

File: Scripts/script-ExtractURL.yml [hunk 3]
```diff
@@ -3,17 +3,35 @@ commonfields:
...
   var whitelist = getCSVListAsArray('Indicators Whitelist');
   while (found = reg.exec(text)) {
-      if (filtered.indexOf(found[0]) < 0 && whitelist.indexOf(found[0]) < 0) {
-          matches[found[0]] = true;
+      matches[found[0]] = true;
+      for (var i = 0 ; i < whitelist.length; i++) {
+          if (whitelist[i] === found[0]) {
+              delete matches[found[0]];
+              break;
+          }else{
```
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: should be changed to else if (one line)

File: Scripts/script-ExtractURL.yml [hunk 4]
```diff
@@ -3,17 +3,35 @@ commonfields:
...
+      return false;
+  }
+
   var text = args.text;
   if (typeof text !== 'string') {
       text = JSON.stringify(args.text).replace(/\\n/g,' '); // need to replace \n
   }
   var matches = {}, found;
-  var reg = /(?:(?:https?|ftp):\/\/|www\.|ftp\.)(?:\([-A-Z0-9+&@#\/%=~_|$?!:,.]*\)|[-A-Z0-9+&@#\/%=~_|$?!:,.])*(?:\([-A-Z0-9+&@#\/%=~_|$?!:,.]*\)|[A-Z0-9+&@#\/%=~_|$])/gmi;
+  var reg = /(?:(?:https?|ftp):\/\/|www\.|ftp\.)(?:\([-A-Z0-9+&@#\/@#=@#|$?!:,.]*\)|[-A-Z0-9+&@#\/%=~_|$?!:,.])*(?:\([-A-Z0-9+&@#\/%=~_|$?!:,.]*\)|[A-Z0-9+&@#\/%=~_|$])/gmi;
```
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: what is this change that you did to the regex ?

Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: Don't remember I touched it.
Changed it how it looked before.

File: Scripts/script-ExtractURL.yml (hunk 3 above)
Comment (dem-evgeny) [inline Scripts/script-ExtractURL.yml]: Changed

File: Scripts/script-ExtractURL.yml [hunk 5]
```diff
@@ -3,6 +3,15 @@ commonfields:
   version: -1
 name: ExtractURL
 script: |-
+  var checkRegex = function(reg,str){
+      if(whitelist[i].indexOf('~') === 0){
```
Comment (aviadl) [inline Scripts/script-ExtractURL.yml]: reg[i]

**Comment by dorsha** [comment]
@dem-evgeny - please verify it works and merge.

**Pull request closed without merging**